src/gt4py/cartesian/gtc/debug/debug_codegen.py

```diff
@@ -0,0 +1,354 @@
+        j_min, j_max = self.visit(horizontal_mask.j, var="j")
+        conditions = []
+        if i_min is not None:
+            conditions.append(f"({i_min}) <= i")
+        if i_max is not None:
+            conditions.append(f"i < ({i_max})")
+        if j_min is not None:
+            conditions.append(f"({j_min}) <= j")
+        if j_max is not None:
+            conditions.append(f"j < ({j_max})")
+        assert len(conditions)
+        if_code = f"if( {' and '.join(conditions)} ):"
+        self.body.append(if_code)
+
+    def visit_HorizontalInterval(self, horizontal_interval: HorizontalInterval, **kwargs):
+        return self.visit(
+            horizontal_interval.start, **kwargs
+        ) if horizontal_interval.start else None, self.visit(
+            horizontal_interval.end, **kwargs
+        ) if horizontal_interval.end else None
+
+    def visit_HorizontalRestriction(self, horizontal_restriction: HorizontalRestriction, **_):
+        self.visit(horizontal_restriction.mask)
+        self.body.indent()
+        self.visit(horizontal_restriction.body)
+        self.body.dedent()
+
+    def visit_VerticalLoop(self):
+        pass
+
+    def visit_ScalarAccess(self, scalar_access: ScalarAccess, **_):
+        return scalar_access.name
+
+    def visit_ScalarDecl(self, scalar_declaration: ScalarDecl, **_) -> str:
+        return scalar_declaration.name
+
+    def visit_NativeFuncCall(self, native_function_call: NativeFuncCall, **_) -> str:
+        arglist = [self.visit(arg) for arg in native_function_call.args]
+        arguments = ",".join(arglist)
+        return f"ufuncs.{native_function_call.func.value}({arguments})"
+
+    def visit_UnaryOp(self, unary_operator: UnaryOp, **_) -> str:
+        return unary_operator.op.value + " " + self.visit(unary_operator.expr)
+
+    def visit_TernaryOp(self, ternary_operator: TernaryOp, **_) -> str:
+        return f"{self.visit(ternary_operator.true_expr)} if {self.visit(ternary_operator.cond)} else {self.visit(ternary_operator.false_expr)}"
+
+    def visit_MaskStmt(self, mask_statement: MaskStmt, **_):
+        self.body.append(f"if {self.visit(mask_statement.mask)}:")
+        with self.body.indented():
+            for statement in mask_statement.body:
+                self.visit(statement)
+
+    def visit_LocalScalar(self, local_scalar: LocalScalar, **__) -> None:
+        raise NotImplementedError(
+            "This state should not be reached because LocalTemporariesToScalars should not have been called."
+        )
+
+    def visit_CacheDesc(self, cache_descriptor: CacheDesc, **_):
+        raise NotImplementedError("Caches should never be visited in the debug backend")
+
+    def visit_IJCache(self, ij_cache: IJCache, **_):
+        raise NotImplementedError("Caches should never be visited in the debug backend")
+
+    def visit_KCache(self, k_cache: KCache, **_):
+        raise NotImplementedError("Caches should never be visited in the debug backend")
```

Comment:
make these differentiable such  that - in case of error - it's clear which one triggered.

```suggestion
    def visit_CacheDesc(self, cache_descriptor: CacheDesc, **_):
        raise NotImplementedError("CacheDescriptors should never be visited in the debug backends")

    def visit_IJCache(self, ij_cache: IJCache, **_):
        raise NotImplementedError("IJCaches should never be visited in the debug backend.")

    def visit_KCache(self, k_cache: KCache, **_):
        raise NotImplementedError("KCaches should never be visited in the debug backend.")
```